src/main/java/org/apache/flink/graph/streaming/library/ConnectedComponents.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.graph.streaming.library;
+
+import org.apache.flink.api.common.functions.GroupReduceFunction;
+import org.apache.flink.api.common.functions.ReduceFunction;
+import org.apache.flink.graph.streaming.EdgesFold;
+import org.apache.flink.graph.streaming.GraphWindowStream;
+import org.apache.flink.graph.streaming.WindowGraphAggregation;
+import org.apache.flink.graph.streaming.example.IterativeConnectedComponents;
+import org.apache.flink.graph.streaming.example.util.DisjointSet;
+import org.apache.flink.types.NullValue;
+
+import java.io.Serializable;
+
+/**
+ * The Connected Components algorithm assigns a component ID to each vertex in the graph.
+ * Vertices that belong to the same component have the same component ID.
+ * This algorithm computes _weakly_ connected components, i.e. edge direction is ignored.
+ * <p>
+ * This is a single-pass implementation, which uses a {@link WindowGraphAggregation} to periodically merge
+ * the partitioned state. For an iterative implementation, see {@link IterativeConnectedComponents}.
+ *
+ * @param <K>  the vertex ID type
+ * @param <EV> the edge value type
+ */
+public class ConnectedComponents<K extends Serializable, EV> extends WindowGraphAggregation<K, EV, DisjointSet<K>, DisjointSet<K>> implements Serializable {
+
+	private long mergeWindowTime;
+
+	public ConnectedComponents(long mergeWindowTime) {
```

Comment:
changed to :
	 Creates a ConnectedComponents object using WindowGraphAggregation class.
	 This helps dividing the edge stream into parallel window for each partition.
	 @param mergeWindowTime Window time in millisec for the merger.